versioned/storage/cassandra/src/main/java/org/projectnessie/versioned/storage/cassandra/CassandraBackend.java

```diff
@@ -329,16 +334,40 @@ boolean executeCas(BoundStatement stmt) {
   }
 
   ResultSet execute(BoundStatement stmt) {
-    return session.execute(stmt);
+    try {
+      return session.execute(stmt);
+    } catch (DriverException e) {
+      throw unhandledException(e);
+    }
   }
 
   CompletionStage<AsyncResultSet> executeAsync(BoundStatement stmt) {
     return session.executeAsync(stmt);
   }
 
-  void handleDriverException(DriverException e) {
-    if (e instanceof CASWriteUnknownException) {
-      logCASWriteUnknown((CASWriteUnknownException) e);
+  static RuntimeException unhandledException(DriverException e) {
+    if (e instanceof QueryExecutionException) {
+      return new UnknownOperationResultException(e);
+    } else if (e instanceof DriverTimeoutException
+        || e instanceof NodeUnavailableException
+        || e instanceof AllNodesFailedException) {
```

Comment:
In theory you should iterate over `getAllErrors()` and recursively call this method for each one of them.

Reply:
Is it worth it?

Reply:
I think so? This is the error that is surfaced to clients when retries happened behind the scenes. For example, `CASWriteUnknownException` is usually retried, so you normally don't encounter it directly, but rather, nested inside `AllNodesFailedException`.

(that said, I think we don't retry a ton of things... the default idempotence is false – I will look into that.)

Reply:
> In theory you should iterate over getAllErrors() and recursively call this method for each one of them.

(Side rant: this C* driver is overly complicated)